stackslib/src/net/download/nakamoto.rs

```diff
@@ -0,0 +1,4283 @@
+                .block_height_to_reward_cycle(
+                    sortdb.first_block_height,
+                    sort_tip.block_height.saturating_add(1),
+                )
+                .expect("FATAL: burnchain tip is before system start")
+        } else {
+            sortdb
+                .pox_constants
+                .block_height_to_reward_cycle(sortdb.first_block_height, sort_tip.block_height)
+                .expect("FATAL: burnchain tip is before system start")
+        };
+
+        test_debug!(
+            "last_sort_height = {}, sort_rc = {}, next_sort_rc = {}, self.reward_cycle = {}, sort_tip.block_height = {}",
+            last_sort_height,
+            sort_rc,
+            next_sort_rc,
+            self.reward_cycle,
+            sort_tip.block_height,
+        );
+
+        if sort_rc == next_sort_rc {
+            // not at a reward cycle boundary, os just extend self.wanted_tenures
+            test_debug!("Extend wanted tenures since no sort_rc change and we have tenure data");
+            self.extend_wanted_tenures(network, sortdb)?;
+            self.update_tenure_start_blocks(chainstate)?;
+            return Ok(());
+        }
+
+        let can_advance_wanted_tenures =
+            if let Some(prev_wanted_tenures) = self.prev_wanted_tenures.as_ref() {
+                !Self::have_unprocessed_tenures(
+                    sortdb
+                        .pox_constants
+                        .block_height_to_reward_cycle(
+                            self.nakamoto_start_height,
+                            sortdb.first_block_height,
+                        )
+                        .expect("FATAL: nakamoto starts before system start"),
+                    &self.tenure_downloads.completed_tenures,
+                    prev_wanted_tenures,
+                    &self.tenure_block_ids,
+                    &sortdb.pox_constants,
+                    sortdb.first_block_height,
+                    invs.inventories.values(),
+                )
+            } else {
+                test_debug!("No prev_wanted_tenures yet");
+                true
+            };
+        if !can_advance_wanted_tenures {
+            return Ok(());
+        }
+
+        // crossed reward cycle boundary
+        let mut new_wanted_tenures = vec![];
+        Self::update_wanted_tenures_for_reward_cycle(
+            sort_rc + 1,
+            sort_tip,
+            sortdb,
+            &mut new_wanted_tenures,
+        )?;
+
+        let mut new_prev_wanted_tenures = vec![];
+        Self::update_wanted_tenures_for_reward_cycle(
+            sort_rc,
+            sort_tip,
+            sortdb,
+            &mut new_prev_wanted_tenures,
+        )?;
+
+        test_debug!("new_wanted_tenures is now {:?}", &new_wanted_tenures);
+        test_debug!(
+            "new_prev_wanted_tenures is now {:?}",
+            &new_prev_wanted_tenures
+        );
+
+        self.prev_wanted_tenures = if new_prev_wanted_tenures.len() > 0 {
```

Comment:
Similarly should either reverse the if statement to use is_empty or keep as is and use !is_empty()

Reply:
Sure